### PR TITLE
Extract gRPC common stuff

### LINKED ISSUE #12602: Extract grpc-common for Quarkiverse Google Cloud Services
**Description**
Multiple Google Cloud Services that are available as extensions inside the Quarkiverse Google Cloud Services project relies on gRPC to communicate with their backend.

The current implementation import the `quarkus-grpc` extension.
It works but it's not ideal as it has a lot of dependencies and setup a lot of un-needed gRPC related functionalities.

The idea is to create an un-listed extension `quarkus-grpc-common` that will include the minimal stuff needed to make gRPC run inside Quarkus.
This should mainly be dependency management and GraalVM stuff

### PR DESCRIPTION
**Pull request by loicmathieu**
Fixes #12602

This externalize the gRPC dependency management and native image stuff so it can be share with the Quarkiverse Google Cloud Services extension.

### CODE REVIEW

File: extensions/grpc-common/deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-grpc-common-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-grpc-common-deployment</artifactId>
```
Comment (gsmet) [inline extensions/grpc-common/deployment/pom.xml:12]: We need a name. Please use the usual convention.

File: extensions/grpc-common/deployment/pom.xml [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+    <dependencies>
+        <dependency>
+            <groupId>io.quarkus</groupId>
+            <artifactId>quarkus-grpc-common</artifactId>
+            <version>999-SNAPSHOT</version>
```
Comment (gsmet) [inline extensions/grpc-common/deployment/pom.xml:18]: This shouldn't be there, it should be set by the BOM.

File: extensions/grpc-common/pom.xml [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    <parent>
+        <artifactId>quarkus-extensions-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-grpc-common-parent</artifactId>
+    <name>Quarkus - gRPC common - parent</name>
```
Comment (gsmet) [inline extensions/grpc-common/pom.xml:14]:
```suggestion
    <name>Quarkus - gRPC Common - Parent</name>
```

File: extensions/grpc-common/runtime/pom.xml [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <parent>
+        <artifactId>quarkus-grpc-common-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-grpc-common</artifactId>
```
Comment (gsmet) [inline extensions/grpc-common/runtime/pom.xml:12]: We need a name here too.

**Review by gsmet — CHANGES_REQUESTED** [review]
Added some comments.

File: extensions/grpc-common/deployment/pom.xml (hunk 1 above)
Comment (loicmathieu) [reply extensions/grpc-common/deployment/pom.xml:12]: Done

File: extensions/grpc-common/deployment/pom.xml (hunk 2 above)
Comment (loicmathieu) [reply extensions/grpc-common/deployment/pom.xml]: Already in the bom, I missed that one, sorry

File: extensions/grpc-common/pom.xml (hunk 3 above)
Comment (loicmathieu) [reply extensions/grpc-common/pom.xml]: Done

File: extensions/grpc-common/runtime/pom.xml (hunk 4 above)
Comment (loicmathieu) [reply extensions/grpc-common/runtime/pom.xml:12]: Done

**Comment by loicmathieu** [comment]
@gsmet I fixed the pom issues, sorry about that I forget to add names and check the poms before pushing it.

**Review by gsmet — APPROVED** [review]

**Comment by gsmet** [comment]
Things are good from my point of view but I will let @cescoffier and @michalszynkiewicz approve and merge it.

File: extensions/grpc-common/pom.xml [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+    <parent>
+        <artifactId>quarkus-extensions-parent</artifactId>
+        <groupId>io.quarkus</groupId>
+        <version>999-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>quarkus-grpc-common-parent</artifactId>
+    <name>Quarkus - gRPC Common - Parent</name>
```
Comment (gsmet) [inline extensions/grpc-common/pom.xml:14]: Ah wait, this doesn't follow our conventions: the Parent shouldn't have a suffix but the Runtime one should:
```
[INFO] Quarkus - Flyway                                                   [pom]
[INFO] Quarkus - Flyway - Runtime                                         [jar]
[INFO] Quarkus - Flyway - Deployment                                      [jar]
```

**Review by gsmet — DISMISSED** [review]

File: extensions/grpc-common/pom.xml (hunk 5 above)
Comment (loicmathieu) [reply extensions/grpc-common/pom.xml]: I copied it from the gRPC module :(
Fixed.

**Review by michalszynkiewicz — APPROVED** [review]

**Pull request merged**